src/common/arrstr.cpp

```diff
@@ -573,3 +579,165 @@ wxArrayString wxSplit(const wxString& str, const wxChar sep, const wxChar escape
 
     return ret;
 }
+
+
+namespace
+{
+    enum wxStringFragmentType
+    {
+        wxFRAGMENT_TYPE_EMPTY = 0,
+        wxFRAGMENT_TYPE_ALPHA = 1,
+        wxFRAGMENT_TYPE_DIGIT = 2
+    };
+
+
+    // ----------------------------------------------------------------------------
+    // wxStringFragment
+    // ----------------------------------------------------------------------------
+    // 
+    // Lightweight object returned by wxString::GetNaturalFragment()
+    // Consists of either a string which represents a number, or a string which contains no numerical digits.
+    // Is used by wxString::CompareNatural()
+    class wxStringFragment
+    {
+    public:
+        wxStringFragment()
+            : type(wxFRAGMENT_TYPE_EMPTY)
+        {}
+
+        wxString             text;
+        double               value;
+        wxStringFragmentType type;
+    };
+
+
+    wxStringFragment GetFragment(wxString &text)
```

Comment:
If possible, please always write the `&` next to to the type name instead of parameter, i.e., `wxString& text`.